### PR TITLE
update plugin tutorial 

### PR DESCRIPTION
**Pull request by SuperRxan**
Signed-off-by: Jianxiang Ran <rxan_embedded@163.com>
update plugin tutorial for english version

### CODE REVIEW

File: docs/en_US/plugins/plugins_tutorial.md [hunk 1]
```diff
@@ -26,46 +26,31 @@ These limitations are relatively strict, and they almost require compiling and r
 
 ## Plugin development 
 
-Developing plugin is generally carried out in the development environment. eKuiper plugins will be deployed to the production environment after passing debugging and running the development environment.
+Developing plugin is generally carried out in the development environment. eKuiper plugins will be deployed to the production environment after passing debugging and running the development environment. Since the limitation of Golang plugin system, here we provide two practical ways for plugin developer to set up environment for eKuiper plugin: **create plugin project inside eKuiper** and **create plugin project outside eKuiper**.
```
Comment (ngjaying) [inline docs/en_US/plugins/plugins_tutorial.md:29]: create **the** plugin project ?

File: docs/en_US/plugins/plugins_tutorial.md [hunk 2]
```diff
@@ -26,46 +26,31 @@ These limitations are relatively strict, and they almost require compiling and r
...
 - Create file mysql.go under the sinks directory
 - Edit file mysql.go for implementing the plugin
-    -  Implement [api.Sink](https://github.com/lf-edge/ekuiper/blob/master/pkg/api/stream.go) interface
+    - Implement [api.Sink](https://github.com/lf-edge/ekuiper/blob/master/pkg/api/stream.go) interface
     - Export Symbol: Mysql. It could be a constructor function so that each rule can instantiate an own mysql plugin instance. Or it could be the struct which means every rule will share a singleton of the plugin. If the plugin has states like the connection, the first approach is preferred.
 - Edit go.mod, add Mysql driver module
+- Build the eKuiper and plugin
 
-The complete source code of mysql.go is as follows:
+### Create plugin project inside eKuiper
```
Comment (ngjaying) [inline docs/en_US/plugins/plugins_tutorial.md:40]: Add "the" ?

File: docs/en_US/plugins/plugins_tutorial.md [hunk 3]
```diff
@@ -26,46 +26,31 @@ These limitations are relatively strict, and they almost require compiling and r
...
 
+When user develop the plugin by this way, he must firstly download the eKuiper source code and run `make` command inside the eKuiper root directory. There are some plugin examples in the *extensions* directory of the eKuiper project source code. The advantage is all existing official plugins are developed by this way so new plugin developer can directly start instead of setting up a new project from scratch.
```
Comment (ngjaying) [inline docs/en_US/plugins/plugins_tutorial.md:42]: When users develop or when the user develops.  This problem also happens elsewhere, search user to check

**Review by ngjaying — CHANGES_REQUESTED** [review]

File: docs/en_US/plugins/plugins_tutorial.md (hunk 1 above)
Comment (SuperRxan) [reply docs/en_US/plugins/plugins_tutorial.md:29]: ack

File: docs/en_US/plugins/plugins_tutorial.md [hunk 4]
```diff
@@ -164,59 +179,131 @@ The environment of compiling the plugin should be consistent with that of  eKuip
...
        ```
 
-### Docker compile
+#### Docker compile
 
 eKuiper provides different docker images for different purpose. The development docker image should be used for compiling plugins. From 0.4.0, the eKuiper image with tag x.x.x (e.g. `kuiper:0.4.0`) is the development docker image. For 0.3.x, eKuiper image with tag x.x.x-dev (e.g. `kuiper:0.3.0-dev`) is the development docker image. Compared with the running version, the development version provides the development environment of Go, which lets users compile the plugin that can be completely compatible with the officially published version of eKuiper. The compiling steps in docker are as follows:
 1. Run docker of the development version of eKuiper. Users need to mount the local plugin directory to the directory in docker, and then they can access and compile the plugin project in docker. The author's plugin project is located in the local `/var/git` directory. We map the local directory `/var/git` to the `/home` directory in docker by using the following commands.
     ```go
     docker run -d --name kuiper-dev --mount type=bind,source=/var/git,target=/home lfedge/ekuiper:0.3.0-dev
```
Comment (ngjaying) [inline docs/en_US/plugins/plugins_tutorial.md:241]: Try to use 1.3.0

File: docs/zh_CN/plugins/plugins_tutorial.md [hunk 5]
```diff
@@ -23,100 +23,89 @@ eKuiper 插件机制基于 Go 语言的插件机制，使用户可以构建松
...
 type mysqlConfig struct {
 	Url   string `json:"url"`
 	Table string `json:"table"`
 }
 
 type mysqlSink struct {
 	conf *mysqlConfig
-	//数据库连接实例
-	db   *sql.DB
+	//The db connection instance
```
Comment (ngjaying) [inline docs/zh_CN/plugins/plugins_tutorial.md:70]: Should revert to Chinese

**Review by ngjaying — CHANGES_REQUESTED** [review]

File: docs/zh_CN/plugins/plugins_tutorial.md (hunk 5 above)
Comment (SuperRxan) [reply docs/zh_CN/plugins/plugins_tutorial.md:70]: ack

File: docs/zh_CN/plugins/plugins_tutorial.md [hunk 6]
```diff
@@ -240,9 +323,9 @@ eKuiper 生产环境和开发环境如果不同，开发的插件需要重新编
 
 ### 插件编译
 
-插件原则上应该与生产环境 eKuiper 采用相同环境进行编译。假设生产环境为 eKuiper docker，则应当采用与生产环境相同版本的 dev docker 环境编译插件。例如，生产环境采用 [lfedge/ekuiper:0.3.0](https://registry.hub.docker.com/layers/lfedge/ekuiper/0.3.0/images/sha256-0e3543d33f6f8c56de044d5ff001fd39b9e26f82219ca5fd25605953ed33580e?context=explore)的 docker 镜像，则插件需要在[lfedge/ekuiper:0.3.0-dev](https://registry.hub.docker.com/layers/lfedge/ekuiper/0.3.0-dev/images/sha256-a309d3821b55b01dc01c4f4a04e83288bf5526325f0073197387f2ca425260d0?context=explore) 的环境中进行编译。
+插件原则上应该与生产环境 eKuiper 采用相同环境进行编译。假设生产环境为 eKuiper docker，则应当采用与生产环境相同版本的 dev docker 环境编译插件。例如，生产环境采用 [lfedge/ekuiper:0.4.0-alpine](https://registry.hub.docker.com/layers/lfedge/ekuiper/0.4.0-alpine/images/sha256-f79e9afd020a05f443d1864ee08007fe472e0d15e266d48a1f636fbd0343d507?context=explore) 的docker镜像，则插件需要在[lfedge/ekuiper:0.4.0](https://registry.hub.docker.com/layers/lfedge/ekuiper/0.4.0/images/sha256-dcc1420cbbd501aedd1bfe4093818a69726de1d6365974b69e99e1d5bc671836?context=explore) 的环境中进行编译。
```
Comment (ngjaying) [inline docs/zh_CN/plugins/plugins_tutorial.md:326]: -alpine is wrong. Should be 0.4.0-slim

**Review by ngjaying — CHANGES_REQUESTED** [review]

**Review by ngjaying — APPROVED** [review]

**Pull request merged**